### ISSUE TITLE
Exotic Critters causes Repurposed Structure's nether structures to hang/deadlock

### ISSUE THREAD

**Issue by TelepathicGrunt** [TelepathicGrunt/RepurposedStructures#100]
https://www.curseforge.com/minecraft/mc-mods/exotic-critters

This issue report is a warning to anyone trying to use my mod with Exotic Critters mod which is an mcreator mod. When attempting to get near an Nether City or any of the other structures with mobs at structure generation, Exotic Critters tries to run code when the Blazes/Wither Skeletons are spawned from my structure and causes the world to freeze/hang/deadlock and you cannot exit the game without killing the game.

There is nothing I can do on my end. This something mcreator or Exotic Critters will have to fix on their end sadly. Hopefully they can fix it soon

**Comment by TelepathicGrunt** [comment]
Thread dump of the thread that is deadlocked by Exotic Critters
```
"Server thread" #70 prio=5 os_prio=0 tid=0x000000003e50b000 nid=0x6720 runnable [0x000000004255d000]
   java.lang.Thread.State: TIMED_WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x00000005d2807b10> (a java.lang.String)
        at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:215)
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_223705_bi(SourceFile:139)
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:129)
        at net.minecraft.world.server.ServerChunkProvider.func_212849_a_(ServerChunkProvider.java:130)
        at net.minecraft.world.World.func_217353_a(World.java:167)
        at net.minecraft.world.IWorldReader.func_217348_a(IWorldReader.java:112)
        at net.minecraft.world.World.func_212866_a_(World.java:163)
        at net.minecraft.world.World.func_175726_f(World.java:159)
        at net.minecraft.world.World.func_175649_E(World.java:1027)
        at net.mcreator.experimentalm.procedures.GuisospawnProcedure.executeProcedure(GuisospawnProcedure.java:75)
        at net.mcreator.experimentalm.procedures.GuisospawnProcedure.onEntitySpawned(GuisospawnProcedure.java:110)
        at net.minecraftforge.eventbus.ASMEventHandler_163_GuisospawnProcedure_onEntitySpawned_EntityJoinWorldEvent.invoke(.dynamic)
        at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:85)
        at net.minecraftforge.eventbus.EventBus$$Lambda$2582/1930704886.invoke(Unknown Source)
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302)
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283)
        at net.minecraft.world.server.ServerWorld.func_217440_f(ServerWorld.java:829)
        at net.minecraft.world.server.ChunkManager.func_219237_a_(ChunkManager.java:592)
        at net.minecraft.world.server.ChunkManager$$Lambda$7695/1627747442.apply(Unknown Source)
        at com.mojang.datafixers.util.Either.lambda$mapLeft$0(Either.java:162)
        at com.mojang.datafixers.util.Either$$Lambda$7696/2054762718.apply(Unknown Source)
        at com.mojang.datafixers.util.Either$Left.map(Either.java:38)
        at com.mojang.datafixers.util.Either.mapLeft(Either.java:162)
        at net.minecraft.world.server.ChunkManager.func_219193_a_(ChunkManager.java:569)
        at net.minecraft.world.server.ChunkManager$$Lambda$7692/1115567462.apply(Unknown Source)
        at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:602)
        at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:577)
        at java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:442)
        at net.minecraft.world.chunk.ChunkTaskPriorityQueueSorter.func_219083_b(SourceFile:58)
        at net.minecraft.world.chunk.ChunkTaskPriorityQueueSorter$$Lambda$7611/539786174.run(Unknown Source)
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144)
        at net.minecraft.world.server.ServerChunkProvider$ChunkExecutor.func_213166_h(ServerChunkProvider.java:514)
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118)
        at net.minecraft.world.server.ServerChunkProvider$ChunkExecutor.func_213168_p(ServerChunkProvider.java:522)
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:127)
        at net.minecraft.world.server.ServerChunkProvider.func_212849_a_(ServerChunkProvider.java:130)
        at net.minecraft.world.World.func_217353_a(World.java:167)
        at net.minecraft.world.IWorldReader.func_217348_a(IWorldReader.java:112)
        at net.minecraft.world.World.func_212866_a_(World.java:163)
        at net.minecraft.entity.Entity.func_226288_n_(Entity.java:2838)
        at net.minecraft.entity.Entity.func_70107_b(Entity.java:358)
        at net.minecraft.entity.Entity.func_242281_f(Entity.java:1135)
        at net.minecraft.entity.Entity.func_70080_a(Entity.java:1122)
        at net.minecraft.network.play.ServerPlayNetHandler.func_175089_a(ServerPlayNetHandler.java:892)
        at net.minecraft.command.impl.TeleportCommand.func_201127_a(TeleportCommand.java:140)
        at net.minecraft.command.impl.TeleportCommand.func_200559_a(TeleportCommand.java:108)
        at net.minecraft.command.impl.TeleportCommand.func_198807_f(TeleportCommand.java:42)
        at net.minecraft.command.impl.TeleportCommand$$Lambda$7124/167063223.run(Unknown Source)
        at com.mojang.brigadier.CommandDispatcher.execute(CommandDispatcher.java:262)
        at net.minecraft.command.Commands.func_197059_a(Commands.java:217)
        at net.minecraft.network.play.ServerPlayNetHandler.func_147361_d(ServerPlayNetHandler.java:1119)
        at net.minecraft.network.play.ServerPlayNetHandler.func_244548_c(ServerPlayNetHandler.java:1102)
        at net.minecraft.network.play.ServerPlayNetHandler.func_147354_a(ServerPlayNetHandler.java:1081)
        at net.minecraft.network.play.client.CChatMessagePacket.func_148833_a(SourceFile:36)
        at net.minecraft.network.play.client.CChatMessagePacket.func_148833_a(SourceFile:9)
        at net.minecraft.network.PacketThreadUtil.func_225383_a(SourceFile:21)
        at net.minecraft.network.PacketThreadUtil$$Lambda$8281/1901368045.run(Unknown Source)
        at net.minecraft.util.concurrent.TickDelayedTask.run(SourceFile:18)
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144)
        at net.minecraft.util.concurrent.RecursiveEventLoop.func_213166_h(SourceFile:23)
        at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:734)
        at net.minecraft.server.MinecraftServer.func_213166_h(MinecraftServer.java:159)
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118)
        at net.minecraft.server.MinecraftServer.func_213205_aW(MinecraftServer.java:717)
        at net.minecraft.server.MinecraftServer.func_213168_p(MinecraftServer.java:711)
        at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213161_c(SourceFile:127)
        at net.minecraft.server.MinecraftServer.func_213202_o(MinecraftServer.java:697)
        at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:646)
        at net.minecraft.server.MinecraftServer.func_240783_a_(MinecraftServer.java:232)
        at net.minecraft.server.MinecraftServer$$Lambda$7497/119525506.run(Unknown Source)
        at java.lang.Thread.run(Thread.java:745)

```

**Comment by TelepathicGrunt** [comment]
I left a comment on Exotic Critters's mod page so hopefully a fix on their end can be done soon!

**Comment by P3rf3ctXZer0** [comment]
That is way more effort than I would have done as it is the mods devs or pack devs responsibility to open to feedback but glad I am not the only one who cares enough to pay attention :)

**Comment by TelepathicGrunt** [comment]
exotic critters fixed this in their Exotic Critters 1.16 Experimental 1.10 update now! closing report
